base/uk.ac.stfc.isis.ibex.scriptgenerator/defined_actions/action_loader.py

```diff
@@ -0,0 +1,72 @@
+from py4j.clientserver import ClientServer, JavaParameters, PythonParameters
+from py4j.java_collections import ListConverter
+import inspect
+import argparse
+from action_interface import Action
+import os
+import sys
+
+
+class Config(object):
+    class Java:
+        implements = ['uk.ac.stfc.isis.ibex.scriptgenerator.pythoninterface.Config']
+
+    def __init__(self, name: str, action: Action):
+        self.action = action
+        self.name = name
+
+    def getName(self) -> str:
+        return self.name
+
+    def getParameters(self):
+        arguments = inspect.getfullargspec(self.action.run).annotations
+        return ListConverter().convert(arguments.keys(), gateway._gateway_client)
+
+    def doAction(self, list_of_arguments):
+        self.action.run(*list_of_arguments)
+
+    def parametersValid(self, list_of_arguments):
+        return self.action.parameters_valid(*list_of_arguments)
+
+    def equals(self, other_config):
```

Comment:
We are implementing an equals method should we also implement a hashCode method?